scripts/export_demucs_onnx.py

```
@@ -1,260 +1,438 @@
+        phase = 2.0 * math.pi * k.unsqueeze(1) * nn.unsqueeze(0) / n_fft
+        win = torch.hann_window(win_length)
+        if win_length < n_fft:
+            pl  = (n_fft - win_length) // 2
+            win = F.pad(win, (pl, n_fft - win_length - pl))
+        # Pre-bake window into filter weights → [freq, 1, n_fft]
+        cos_filt = (torch.cos(phase) * win.unsqueeze(0)).unsqueeze(1)
+        sin_filt = (torch.sin(phase) * win.unsqueeze(0)).unsqueeze(1)
+        self.register_buffer('cos_filt', cos_filt)   # [freq, 1, n_fft]
+        self.register_buffer('sin_filt', sin_filt)   # [freq, 1, n_fft]
+
+    def forward(self, x, pad_fft):
```

Ruff (0.15.12)

[warning] Missing return type annotation for private function `forward`

(ANN202)